### ISSUE TITLE
Incompatible Format errors on Andriod/Chrome with Depth Texture extension

### ISSUE THREAD

**Issue by bhouston** [mrdoob/three.js#8597]
##### Description of the problem

@MasterJames reported this error with his Galaxy Note 3 here:

> I seem to have to rerun adb kill-server and adb start-server for chrome://inspect to detect my device with is actually a Gallaxy Note 3.
> 
> THREE.WebGLRenderer 76dev
> three.js:28971 THREE.WebGLRenderer: OES_texture_float_linear extension not supported.
> three.js:31660 WebGL: INVALID_VALUE: texImage2D: invalid internalformat
> three.js:25244 WebGL: INVALID_FRAMEBUFFER_OPERATION: clear: the internalformat of the attached image is not depth-renderable
> three.js:28805 WebGL: INVALID_FRAMEBUFFER_OPERATION: drawArrays: the internalformat of the attached image is not depth-renderable

[Source](https://github.com/mrdoob/three.js/pull/8577#issuecomment-208056923)

This may be related to the stencil buffer incompatibility with the current Depth Texture implementation reported by @kintel here: https://github.com/mrdoob/three.js/issues/8596
##### Three.js version
- [x] Dev
- [ ] r75
- [ ] ...
##### Browser
- [ ] All of them
- [x] Chrome
- [ ] Firefox
- [ ] Internet Explorer
##### OS
- [ ] All of them
- [ ] Windows
- [ ] Linux
- [x] Android
- [ ] IOS

**Comment by bhouston** [comment]
My theory is that the reason it is crashing is because it is trying to use a Float format with linear interpolation for the back buffer that is not supported.

@MasterJames, can you paste the screenshots or list of supported extensions in Chrome and Firefox you are seeing on your Galaxy Note 3 via this website?

http://renderingpipeline.com/webgl-extension-viewer/

**Comment by MasterJames** [comment]
```
Your WebGL implementation:

Renderer: WebKit WebGL
Vendor: WebKit
WebGL version: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL version: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Supported WebGL extensions:

ANGLE_instanced_arrays
EXT_blend_minmax
EXT_disjoint_timer_query
EXT_sRGB
EXT_texture_filter_anisotropic
WEBKIT_EXT_texture_filter_anisotropic
OES_element_index_uint
OES_standard_derivatives
OES_texture_float
OES_texture_half_float
OES_texture_half_float_linear
OES_vertex_array_object
WEBGL_compressed_texture_atc
WEBKIT_WEBGL_compressed_texture_atc
WEBGL_compressed_texture_etc1
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBKIT_WEBGL_lose_context
Implementation dependent limits:

WebGL constant	minimum	maximum on your browser
MAX_COMBINED_TEXTURE_IMAGE_UNITS	8	32
MAX_CUBE_MAP_TEXTURE_SIZE	16	4096
MAX_FRAGMENT_UNIFORM_VECTORS	16	224
MAX_RENDERBUFFER_SIZE	1	4096
MAX_TEXTURE_IMAGE_UNITS	8	16
MAX_TEXTURE_SIZE	64	4096
MAX_VARYING_VECTORS	8	16
MAX_VERTEX_ATTRIBS	8	16
MAX_VERTEX_TEXTURE_IMAGE_UNITS	0	16
MAX_VERTEX_UNIFORM_VECTORS	128	256
MAX_VIEWPORT_DIMS	see specs	(4096, 4096)
NUM_COMPRESSED_TEXTURE_FORMATS	0	null

```

**Comment by MasterJames** [comment]
Through Firefox on the Note 3 I get...
`
Your WebGL implementation:

    Renderer: Mozilla
    Vendor: Mozilla
    WebGL version: WebGL 1.0
    GLSL version: WebGL GLSL ES 1.0

Supported WebGL extensions:

ANGLE_instanced_arrays
EXT_blend_minmax
EXT_color_buffer_half_float
EXT_frag_depth
EXT_sRGB
EXT_texture_filter_anisotropic
OES_element_index_uint
OES_standard_derivatives
OES_texture_float
OES_texture_float_linear
OES_texture_half_float
OES_texture_half_float_linear
OES_vertex_array_object
WEBGL_color_buffer_float
WEBGL_compressed_texture_atc
WEBGL_compressed_texture_etc1
WEBGL_depth_texture
WEBGL_lose_context
MOZ_WEBGL_lose_context
MOZ_WEBGL_compressed_texture_atc
MOZ_WEBGL_depth_texture

Implementation dependent limits:

WebGL constant	minimum	maximum on your browser
MAX_COMBINED_TEXTURE_IMAGE_UNITS	8	32
MAX_CUBE_MAP_TEXTURE_SIZE	16	4096
MAX_FRAGMENT_UNIFORM_VECTORS	16	224
MAX_RENDERBUFFER_SIZE	1	4096
MAX_TEXTURE_IMAGE_UNITS	8	16
MAX_TEXTURE_SIZE	64	4096
MAX_VARYING_VECTORS	8	16
MAX_VERTEX_ATTRIBS	8	16
MAX_VERTEX_TEXTURE_IMAGE_UNITS	0	16
MAX_VERTEX_UNIFORM_VECTORS	128	256
MAX_VIEWPORT_DIMS	see specs	(4096, 4096)
NUM_COMPRESSED_TEXTURE_FORMATS	0	null
`

**Comment by mattdesl** [comment]
Are you using Chrome or the stock Android browser? Your first report doesn't show WEBGL_depth_texture support. However, the demo should show a red error box stating that, rather than crash with a black screen. 

I tested in Chrome on our Samsung Note 4 and I got the red error box stating depth textures are not supported.

**Comment by MasterJames** [comment]
Yes I posted the version before Chrome 49.0.2623.105
Must be hardware drivers. It's Android 5.0.

**Comment by bhouston** [comment]
It shouldn't be hardware drivers if it is just an error message that should
display...  the extension isn't there, the demo should just show a black
screen with an error message.   This should be easy to get right I would
think with no error messages.

Best regards,
Ben Houston (Cell: 613-762-4113, Skype: ben.exocortex, Twitter:
@exocortexcom)
https://Clara.io - Online 3D Modeling and Rendering

**Comment by MasterJames** [comment]
Okay I can try to check. Where is the code that should be popping up the error?  ( Oh you probably want to delete your cell above. )

**Comment by n3ssi3** [comment]
Any news on that issue? I am having the same problem on my Samsung Galaxy J7 Android 5.1.1
Both browsers Goolge Chrome and android stack browser are not working :(

**Comment by Mugen87** [comment]
Should be fixed via #9368. If devices still struggle with depth textures, please file a new issue. The  [webgl_depth_texture](https://threejs.org/examples/webgl_depth_texture) demo allows now to configure the depth texture via `dat.gui` which makes it easier to verify configurations.
